CodegenProposal.md

```diff
@@ -338,13 +338,142 @@ A `SchemaConfiguration` object will also be generated for your schema. This obje
+
+```swift
+enum RelativeSize: String, EnumType {
+  case LARGE
+  case AVERAGE
+  case SMALL
+}
+```
+```swift
+struct Animal: SelectionSet {
+  // ...
+  var size: GraphQLEnum<RelativeSize> { data["size"] }
+}
+```
+`GraphQLEnum` wraps your generated `EnumType`s and provides the `__unknown` case with an associated value of a raw string. This is necessary for clients to provide forward-compatibility with new enum cases added to a schema in the future. `GraphQLEnum` has pattern matching and `Equatable` conformance implemented that allows you to consume it as if it were your underlying `EnumType` in most cases. 
+
+**Examples:**
+```swift
+let size: GraphQLEnum<RelativeSize> = .init(.SMALL)
+
+size == .SMALL // true
+```
+When using switch, you must provide a case for the unknown value. 
+```swift
+switch size {
+case .SMALL: break
+case .AVERAGE: break
+case .LARGE: break
+case .__unknown(_): break
+default: break
+}
+```
+Because pattern matching is being used to match against the underling `EnumType` cases, you must also provide a default case.
+
+To ensure exhaustive switch cases without a default case your generated cases can be wrapped in `.case()`.
+```swift
+switch size {
+case .case(.SMALL): break
+case .case(.AVERAGE): break
+case .case(.LARGE): break
+case .__unknown(_): break
+}
+```
+If you want to ignore the unknown case, you can access the `.value` field, which returns an optional value of the wrapped type. If the type is an unknown case `.value` will be `nil`.
+```swift
+switch size.value {
+case .SMALL: break
+case .AVERAGE: break
+case .LARGE: break
+default: break 
+// or
+case .none: break
+}
+```
+See [GraphQLEnum.swift](Sources/ApolloAPI/GraphQLEnum.swift) for implementation details.
+
 # `InputObject` Generation
 
-TODO
+Input objects will be generated for each `input` type in the schema that is used in an argument for any of the operations defined in your application. Input objects are structs that are backed by a `InputDict` struct that stores the values for the fields on the input object in a dictionary. This allows for `InputObject`s to be treated as values types but use copy-on-write semantics under the hood.
 
-# `EnumType` Generation
+Nullable fields on input objects are represented using `GraphQLNullable` to allow for both `null` and `nil` values. 
 
-TODO
+Following the [Input Coercion rules](https://spec.graphql.org/draft/#sec-Input-Objects.Input-Coercion) from the GraphQL spec, the server defined default value for a field will be used when passing `nil`. For non-nullable fields, if the schema provides a default value, the field will be represented as an optional to allow for this. Nullable fields on input objects are represented using `GraphQLNullable` to allow for both `null` and `nil` values.
```

Comment:
One thing probably worth clarifying here: We're not going to send what's in the schema at the time the code was generated, we just won't send anything, which will cause the current default on the server to be used. 

This is useful if the default value on the server side has changed since the code was originally generated

Reply:
Good point. I'll add this now, thanks.